pdns/tcpreceiver.cc

```diff
@@ -544,26 +544,20 @@ int TCPNameserver::doAXFR(const DNSName &target, std::unique_ptr<DNSPacket>& q,
   bool securedZone = dk.isSecuredZone(target);
   bool presignedZone = dk.isPresigned(target);
 
-  bool noAXFRBecauseOfNSEC3Narrow=false;
   NSEC3PARAMRecordContent ns3pr;
   bool narrow;
   bool NSEC3Zone=false;
   if(securedZone && dk.getNSEC3PARAM(target, &ns3pr, &narrow)) {
     NSEC3Zone=true;
     if(narrow) {
-      g_log<<Logger::Error<<"Not doing AXFR of an NSEC3 narrow zone '"<<target<<"' for "<<q->getRemote()<<endl;
-      noAXFRBecauseOfNSEC3Narrow=true;
+      g_log<<Logger::Warning<<"Not doing AXFR of an NSEC3 narrow zone '"<<target<<"' for "<<q->getRemote()<<endl;
+      outpacket->setRcode(RCode::Refused);
+      // FIXME: should actually figure out if we are auth over a zone, and send out 9 if we aren't
```

Comment:
I thought 9 was reserved for RFC2136

Reply:
There is nothing to figure out here. The zone is narrow so it does exist. I will remove the comment.

Reply:
ah yes, in this context there is nothing to figure out. Thanks :)